### PR TITLE
Multi-Window support added for android N

### PR DESCRIPTION
**Pull request by GurpreetSK95**
device-2016-07-20-155248

### CODE REVIEW

**Comment by AndyScherzinger** [comment]
Hi @GurpreetSK95 can you also update the travis build configuration since the update to 24 breaks the actual builds (https://travis-ci.org/owncloud/android/builds/146066120) since the build config needs to be updated too :smiley:

**Comment by GurpreetSK95** [comment]
Hi @AndyScherzinger and thanks for the inputs. I've made the changes to .travis.yml file and hope the request passes the checks this time. Regards.

**Comment by AndyScherzinger** [comment]
Hi @GurpreetSK95 unfortunately I doesn't :(

--> ```Error: Invalid --abi armeabi-v7a for the selected target.```
--> see: https://travis-ci.org/owncloud/android/builds/146159491

**Comment by GurpreetSK95** [comment]
Sorry for that @AndyScherzinger. I wanted to ask whether the java version "1.7.0_76" might be causing some issue, as android N requires java version "1.8.0_91"

**Comment by AndyScherzinger** [comment]
Hi @GurpreetSK95 
I am not sure if that might also cause some issues down the road in the build. Right now it fails right at the start since the referenced device/image can't be found.

File: build.gradle [hunk 1]
```diff
@@ -77,6 +77,11 @@ android {
         testInstrumentationRunnerArgument "TEST_USER", "\"$System.env.OCTEST_APP_USERNAME\""
         testInstrumentationRunnerArgument "TEST_PASSWORD", "\"$System.env.OCTEST_APP_PASSWORD\""
         testInstrumentationRunnerArgument "TEST_SERVER_URL", "\"$System.env.OCTEST_SERVER_BASE_URL\""
+
+        jackOptions {
+            enabled true
```
Comment (davivel) [inline build.gradle]: What is this for?

Comment (davivel) [inline build.gradle]: Oh, the new Java compiler. This is strictly needed, if build breaks due to it we could remove it (but that's not happenning right now) . ReferencE: https://developer.android.com/preview/setup-sdk.html#get-as13

**Comment by davivel** [comment]
Hi, @GurpreetSK95 ; thanks a lot for your contribution :)

Try to add 
```
jdk:
  - oraclejdk8
```
right before the ```android:``` line in travis.yml to get a JDK 1.8 environment in the build.

I'll have a look to see what emulator image can be used instead of the one failing now.

Meanwhile, could you sign the contributor agreement in https://owncloud.org/contribute/agreement/ and send it to cla@owncloud.org?

Thanks!

**Comment by davivel** [comment]
Oh, the emulator image is downloading correctly, no problem with that.

But the ownCloud library for Android is not updated to android-24, that's making the build fail. You'll need to do a new PR to https://github.com/owncloud/android-library updating it's build options, and then add a commit here to update the reference to the library.

File: build.gradle (hunk 1 above)
Comment (GurpreetSK95) [inline build.gradle]: Hi @davivel. This was added as the compiler was giving an error "Error:Jack is required to support java 8 language features. Either enable Jack or remove sourceCompatibility JavaVersion.VERSION_1_8".  More [here](http://stackoverflow.com/questions/37004069/errorjack-is-required-to-support-java-8-language-features).

**Comment by davivel** [comment]
Hi, @GurpreetSK95 . Now we can resume the work here. 

3 things left:
 - [ ] sign licence/cla
 - [ ] rebase ```GurpreetSK95:ChangesForAndroidN``` on ```owncloud:master```
 - [ ] update reference of submodule owncloud-android-library so that it points to ```owncloud/android-library:master````

For the rebase, you need enter the root folder of the local copy of your fork and apply:

```
git fetch upstream
git checkout ChangesForAndroidN
git rebase upstream/master
git push -f origin ChangesForAndroidN
```

Then, for the third point follow with:

```
git checkout ChangesForAndroidN
cd owncloud-android-library
git checkout master
git pull
cd ..
git add .
git commit -m "Updated reference to OC library"
git push origin ChangesForAndroidN
```

Ping me if anything goes wrong.

**Comment by davivel** [comment]
@GurpreetSK95 , if you don't mind I will rebase by myself.

**Comment by davivel** [comment]
Closing to rebase in a branch in this repo...

**Pull request closed without merging**